### PR TITLE
Bump golang from 1.18 to 1.19

### PR DESCRIPTION
**Pull request by Ashmita152**
Signed-off-by: Ashmita152 <ashmita.bohara@logz.io>

## Which problem is this PR solving?
-  Bump golang from 1.18 to 1.19
- Bump alpine from 3.14 to 3.16

## Short description of the changes
- Also added `ReadHeaderTimeout` to fix linter failures
```G112: Potential Slowloris Attack because ReadHeaderTimeout is not configured in the http.Server (gosec)```

### CODE REVIEW

File: cmd/agent/app/httpserver/srv.go [hunk 1]
```diff
@@ -39,8 +40,9 @@ func NewHTTPServer(hostPort string, manager configmanager.ClientConfigManager, m
 	handler.RegisterRoutes(r)
 	errorLog, _ := zap.NewStdLogAt(logger, zapcore.ErrorLevel)
 	return &http.Server{
-		Addr:     hostPort,
-		Handler:  r,
-		ErrorLog: errorLog,
+		Addr:              hostPort,
+		Handler:           r,
+		ErrorLog:          errorLog,
+		ReadHeaderTimeout: 2 * time.Second,
```
Comment (yurishkuro) [inline cmd/agent/app/httpserver/srv.go:46]: why is this necessary? Doesn't default work?

File: cmd/query/app/flags.go [hunk 2]
```diff
@@ -142,7 +142,8 @@ func (qOpts *QueryOptions) BuildQueryServiceOptions(storageFactory storage.Facto
 }
 
 // stringSliceAsHeader parses a slice of strings and returns a http.Header.
-//  Each string in the slice is expected to be in the format "key: value"
+//
+//	Each string in the slice is expected to be in the format "key: value"
```
Comment (yurishkuro) [inline cmd/query/app/flags.go:146]:
```suggestion
// Each string in the slice is expected to be in the format "key: value"
```

File: cmd/query/app/query_parser.go [hunk 3]
```diff
@@ -101,30 +101,33 @@ func newDurationUnitsParser(units time.Duration) durationParser {
 // parseTraceQueryParams takes a request and constructs a model of parameters.
 //
 // Why start/end parameters are expressed in microseconds:
-//     Span searches operate on span latencies, which are expressed as microseconds in the data model, hence why
-//     support for high accuracy in search query parameters is required.
-//     Microsecond precision is a legacy artifact from zipkin origins where timestamps and durations
-//     are in microseconds (see: https://zipkin.io/pages/instrumenting.html).
+//
+//	Span searches operate on span latencies, which are expressed as microseconds in the data model, hence why
```
Comment (yurishkuro) [inline cmd/query/app/query_parser.go:105]: doesn't this cause code block formatting?

File: pkg/config/string_slice.go [hunk 4]
```diff
@@ -17,9 +17,12 @@ package config
 import "strings"
 
 // StringSlice implements the pflag.Value interface and allows for parsing multiple
-//  config values with the same name
+//
+//	config values with the same name
```
Comment (yurishkuro) [inline pkg/config/string_slice.go:21]: this isn't code block

**Review by yurishkuro — COMMENTED** [review]
a lot of comments are incorrectly indented and will be rendered as code blocks instead of plain text.

File: cmd/agent/app/httpserver/srv.go (hunk 1 above)
Comment (Ashmita152) [reply cmd/agent/app/httpserver/srv.go:46]: Hi Yuri, without it linter was failing with this error ```G112: Potential Slowloris Attack because ReadHeaderTimeout is not configured in the http.Server (gosec)``` .

**Comment by Ashmita152** [comment]
The identation issue is now resolved. The PR is ready for review again. Thank you.

**Review by yurishkuro — APPROVED** [review]
